lib/vrl/compiler/src/vm/machine.rs

```diff
@@ -0,0 +1,466 @@
+use super::{state::VmState, Variable, VmArgumentList};
+use crate::{vm::argument_list::VmArgument, Context, ExpressionError, Function, Value};
+use diagnostic::Span;
+use std::{collections::BTreeMap, ops::Deref};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum OpCode {
+    Abort,
+    Return,
+    Constant,
+    Negate,
+    Add,
+    Subtract,
+    Multiply,
+    Divide,
+    Rem,
+    Merge,
+    Not,
+    Greater,
+    GreaterEqual,
+    Less,
+    LessEqual,
+    NotEqual,
+    Equal,
+    Pop,
+    ClearError,
+    JumpIfFalse,
+    JumpIfTrue,
+    JumpIfNotErr,
+    Jump,
+    SetPathInfallible,
+    SetPath,
+    GetPath,
+    Call,
+    CreateArray,
+    CreateObject,
+    EmptyParameter,
+    MoveParameter,
+    MoveStatic,
```

Comment:
I wasn't sure what this last one meant. We might want to use a better name, or add a comment.